### PR TITLE
[Issue #570] Add a whitelist to let a directory through the media scanner blacklist

### PR DESCRIPTION
**Pull request by spark4160**
These changes add a whitelist setting to the media library settings page that overrides the media scanner's blacklist.

Since d2c54eeb2a395d74cdd2d02ca277d4ad37a0ad1a (I think), my music has not shown up in the library (I've been using the files tab instead). This is because I use syncthing to get my music onto my phone's SD card, but syncthing can only write within its data folder on removable storage (/storage/6BA1-35CD/Android/data/com.nutomic.syncthingandroid/files/Music in my case). This path is blacklisted so vanilla does not add any files inside to the library. Using /storage/6BA1-35CD/Android/data/com.nutomic.syncthingandroid/files/Music as the whitelist setting allows my music to be scanned but should still skip over ringtones etc.

### CODE REVIEW

File: src/ch/blinkenlights/android/medialibrary/MediaSchema.java [hunk 1]
```diff
@@ -302,6 +303,11 @@ public static void upgradeDatabaseSchema(SQLiteDatabase dbh, int oldVersion) {
 			dbh.execSQL(VIEW_CREATE_SONGS_ALBUMS_ARTISTS_HUGE);
 		}
 
+		if (oldVersion < 20170324 && oldVersion >= 20170120) {
+			// Added a string field to allow for string based preferences
+			dbh.execSQL("DROP TABLE preferences");
```
Comment (Adonai) [inline src/ch/blinkenlights/android/medialibrary/MediaSchema.java]: What if somebody had data there? We'll drop it altogether

Comment (spark4160) [inline src/ch/blinkenlights/android/medialibrary/MediaSchema.java]: You're right, and it should be easy enough to just alter the table to add the string column.

**Comment by adrian-bl** [comment]
This looks like a big workaround for a problem which is not ours (but Syncthings & Androids).

I am planning to implement an option to select folders for the mediascanner (like VLC does) which would also solve this issue. So i'd rather spend time to implement this properly than merging a workaround.

(Nb: Android >=7 can use sdcards as 'internal storage' which also solves the syncthing issue)

**Comment by spark4160** [comment]
Fair enough, thanks in advance for when you get that working.

**Comment by adrian-bl** [comment]
Closing this PR as the feature to pick indexable directories has been merged :-)

**Pull request closed without merging**